### PR TITLE
CRYP-11: Setup docker and docker compose

### PR DESCRIPTION
**Pull request by jason-gerard**
## Changes
- Add docker files for
  - client web build
  - api
  - api database
- Add docker-compose file for dev environment

### CODE REVIEW

**Review by alexbara2000 — APPROVED** [review]
It is very good. However can we wait before creating database tables, there still needs to be a conversation about what we are storing and how.

**Comment by jason-gerard** [comment]
> It is very good. However can we wait before creating database tables, there still needs to be a conversation about what we are storing and how.

No database tables are being made here, just a dockerfile for the database itself

File: packages/docker/api.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM node:16-slim AS base
```
Comment (Andre-Ibrahim) [inline packages/docker/api.Dockerfile:1]: 16-slim doesn't work for m1 should we make it 16 to avoid confusion

**Review by Andre-Ibrahim — APPROVED** [review]

File: packages/docker/api.Dockerfile (hunk 1 above)
Comment (jason-gerard) [reply packages/docker/api.Dockerfile:1]: The issue with that is only with the bcrypt library. But I think I found a solution for it we can try when we set it up for that user story.

Comment (Andre-Ibrahim) [reply packages/docker/api.Dockerfile:1]: yea you're right, looking forward to seeing your solution!

**Pull request merged**